src/Plugins/SimpleCommands/CustomCommand.cs

```diff
@@ -1,17 +1,27 @@
-﻿using System;
+﻿using ChatterBot.Core;
+using ChatterBot.Plugins.SimpleCommands.Validation;
+using System;
+using System.ComponentModel;
+using System.Linq;
 
-namespace ChatterBot.Core.SimpleCommands
+namespace ChatterBot.Plugins.SimpleCommands
 {
-    public class CustomCommand : BaseBindable
+    public class CustomCommand : BaseBindable, IDataErrorInfo
     {
         public int Id { get; set; }
 
         private string _commandWord = "!";
         private Access _access = Access.Everyone;
-        private string _response = string.Empty;
-        private int _cooldownTime = 0;
-        private int _userCooldownTime = 0;
+        private string _response;
+        private int _cooldownTime;
+        private int _userCooldownTime;
         private bool _enabled = true;
+        private readonly CustomCommandValidator _validator;
+
+        public CustomCommand()
+        {
+            _validator = new CustomCommandValidator();
```

Comment:
These aren't created by IOC, so cannot DI this easily.

Reply:
Does a plug-in need it's own IOC Container / to be its own composition root?